File: C/Cats_And_Mouse.c
// Cats and a Mouse

#include <stdio.h>
#include <math.h>
#include <string.h>
#include <stdlib.h>

int main()
{
    int n;
    scanf("%d", &n);
    for (int i = 0; i < n; i++)
    {
        int x, y, z;
        scanf("%d %d %d", &x, &y, &z);
        int a = abs(x - z);
        int b = abs(y - z);
        if (a < b)
            printf("Cat A\n");
        else if (b < a)
            printf("Cat B\n");
        else
            printf("Mouse C\n");
    }
    return 0;
}

File: java/Cats_And_Mouse.java
import java.util.Scanner;
import java.lang.Math;
public class Cats_And_Mouse{

    static String catAndMouse(int a,int b,int c){
        if(Math.abs(c-a)==Math.abs(c-b))
        return "Mouse C";
        else if(Math.abs(c-a)<Math.abs(c-b))
        return "Cat A";
        else 
        return "Cat B";
    }

    public static void main(String[] args) {
        Scanner in=new Scanner(System.in);
        int T=in.nextInt();
        while(T!=0){
            --T;
            System.out.println(catAndMouse(in.nextInt(),in.nextInt(),in.nextInt()));
        }
    }
}